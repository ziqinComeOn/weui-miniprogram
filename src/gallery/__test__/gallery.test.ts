import path from 'path'
import simulate from 'miniprogram-simulate'

describe('gallery', () => {
    const gallery = simulate.load(path.resolve(__dirname, '../gallery'), 'mp-gallery')

    const onChange = jest.fn(function (e) {
        this.setData({
            current: e.detail.current
        })
    })

    const onDelete = jest.fn(function (e) {
        this.setData({
            imgUrls: e.detail.currentImgs
        })
    })

    const onHide = jest.fn(function () {
        this.setData({
            show: false
        })
    })

    const id = simulate.load({
        template: `
            <mp-gallery
                id="gallery"
                show="{{show}}"
                bindchange="onChange"
                binddelete="onDelete"
                bindhide="onHide"
                img-urls="{{imgUrls}}"
                delete
                hide-on-click="{{true}}"
                current="{{current}}"
            ></mp-gallery>
        `,
        usingComponents: {
            'mp-gallery': gallery
        },
        data: {
            imgUrls: ['http://qq.com/1.jpg', 'http://qq.com/2.jpg', 'http://qq.com/3.jpg'],
            current: 0,
            show: false
        },
        methods: {
            onChange,
            onDelete,
            onHide
        }
    })

    test('basic', async () => {
        const wrapper = simulate.render(id)
        wrapper.attach(document.createElement('parent-wrapper'))
        const comp = wrapper.querySelector('#gallery')
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()

        wrapper.setData({ show: true })
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()

        wrapper.setData({ current: 1 })
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()

        const deleteBtn = comp.querySelector('.weui-gallery__del')
        deleteBtn.dispatchEvent('tap')
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()
        expect(onDelete).toBeCalledTimes(1)

        const imgWrap = comp.querySelector('.weui-gallery__img__wrp')
        imgWrap.dispatchEvent('tap')
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()
        expect(onDelete).toBeCalledTimes(1)
    })
})
