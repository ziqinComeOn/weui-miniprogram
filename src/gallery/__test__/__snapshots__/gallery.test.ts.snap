// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gallery basic 1`] = `
<mp-gallery
  delete="{{true}}"
  id="gallery"
  bind:change="onChange"
  bind:delete="onDelete"
  bind:hide="onHide"
>
  <wx-view
    class="weui-gallery  "
  >
    <wx-view
      class="weui-gallery__info"
    >
      1/3
    </wx-view>
    <wx-swiper
      autoplay="{{false}}"
      class="weui-gallery__img__wrp"
      current="{{0}}"
      duration="{{500}}"
      indicatorDots="{{false}}"
      bind:change="change"
      bind:tap="hideGallery"
    >
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/1.jpg"
        />
      </swiper-item>
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/2.jpg"
        />
      </swiper-item>
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/3.jpg"
        />
      </swiper-item>
    </wx-swiper>
    <wx-view
      class="weui-gallery__opr"
    >
      <wx-navigator
        class="weui-gallery__del"
        bind:tap="deleteImg"
      >
        删除
      </wx-navigator>
    </wx-view>
  </wx-view>
</mp-gallery>
`;

exports[`gallery basic 2`] = `
<mp-gallery
  delete="{{true}}"
  id="gallery"
  bind:change="onChange"
  bind:delete="onDelete"
  bind:hide="onHide"
>
  <wx-view
    class="weui-gallery weui-gallery_show "
  >
    <wx-view
      class="weui-gallery__info"
    >
      1/3
    </wx-view>
    <wx-swiper
      autoplay="{{false}}"
      class="weui-gallery__img__wrp"
      current="{{0}}"
      duration="{{500}}"
      indicatorDots="{{false}}"
      bind:change="change"
      bind:tap="hideGallery"
    >
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/1.jpg"
        />
      </swiper-item>
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/2.jpg"
        />
      </swiper-item>
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/3.jpg"
        />
      </swiper-item>
    </wx-swiper>
    <wx-view
      class="weui-gallery__opr"
    >
      <wx-navigator
        class="weui-gallery__del"
        bind:tap="deleteImg"
      >
        删除
      </wx-navigator>
    </wx-view>
  </wx-view>
</mp-gallery>
`;

exports[`gallery basic 3`] = `
<mp-gallery
  delete="{{true}}"
  id="gallery"
  bind:change="onChange"
  bind:delete="onDelete"
  bind:hide="onHide"
>
  <wx-view
    class="weui-gallery weui-gallery_show "
  >
    <wx-view
      class="weui-gallery__info"
    >
      2/3
    </wx-view>
    <wx-swiper
      autoplay="{{false}}"
      class="weui-gallery__img__wrp"
      current="{{1}}"
      duration="{{500}}"
      indicatorDots="{{false}}"
      bind:change="change"
      bind:tap="hideGallery"
    >
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/1.jpg"
        />
      </swiper-item>
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/2.jpg"
        />
      </swiper-item>
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/3.jpg"
        />
      </swiper-item>
    </wx-swiper>
    <wx-view
      class="weui-gallery__opr"
    >
      <wx-navigator
        class="weui-gallery__del"
        bind:tap="deleteImg"
      >
        删除
      </wx-navigator>
    </wx-view>
  </wx-view>
</mp-gallery>
`;

exports[`gallery basic 4`] = `
<mp-gallery
  delete="{{true}}"
  id="gallery"
  bind:change="onChange"
  bind:delete="onDelete"
  bind:hide="onHide"
>
  <wx-view
    class="weui-gallery weui-gallery_show "
  >
    <wx-view
      class="weui-gallery__info"
    >
      1/2
    </wx-view>
    <wx-swiper
      autoplay="{{false}}"
      class="weui-gallery__img__wrp"
      current="{{0}}"
      duration="{{500}}"
      indicatorDots="{{false}}"
      bind:change="change"
      bind:tap="hideGallery"
    >
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/1.jpg"
        />
      </swiper-item>
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/3.jpg"
        />
      </swiper-item>
    </wx-swiper>
    <wx-view
      class="weui-gallery__opr"
    >
      <wx-navigator
        class="weui-gallery__del"
        bind:tap="deleteImg"
      >
        删除
      </wx-navigator>
    </wx-view>
  </wx-view>
</mp-gallery>
`;

exports[`gallery basic 5`] = `
<mp-gallery
  delete="{{true}}"
  id="gallery"
  bind:change="onChange"
  bind:delete="onDelete"
  bind:hide="onHide"
>
  <wx-view
    class="weui-gallery  "
  >
    <wx-view
      class="weui-gallery__info"
    >
      2/2
    </wx-view>
    <wx-swiper
      autoplay="{{false}}"
      class="weui-gallery__img__wrp"
      current="{{1}}"
      duration="{{500}}"
      indicatorDots="{{false}}"
      bind:change="change"
      bind:tap="hideGallery"
    >
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/1.jpg"
        />
      </swiper-item>
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://qq.com/3.jpg"
        />
      </swiper-item>
    </wx-swiper>
    <wx-view
      class="weui-gallery__opr"
    >
      <wx-navigator
        class="weui-gallery__del"
        bind:tap="deleteImg"
      >
        删除
      </wx-navigator>
    </wx-view>
  </wx-view>
</mp-gallery>
`;
